packages/dd-trace/src/config.js

```diff
@@ -806,31 +544,285 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
+    this._setUnit(env, 'sampleRate', DD_TRACE_SAMPLE_RATE)
+    this._setValue(env, 'sampler.rateLimit', DD_TRACE_RATE_LIMIT)
+    this._setString(env, 'scope', DD_TRACE_SCOPE)
+    this._setString(env, 'service', DD_SERVICE || DD_SERVICE_NAME || tags.service)
+    this._setString(env, 'site', DD_SITE)
+    if (DD_TRACE_SPAN_ATTRIBUTE_SCHEMA) {
+      this._setString(env, 'spanAttributeSchema', validateNamingVersion(DD_TRACE_SPAN_ATTRIBUTE_SCHEMA))
+    }
+    this._setBoolean(env, 'spanRemoveIntegrationFromService', DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED)
+    this._setBoolean(env, 'startupLogs', DD_TRACE_STARTUP_LOGS)
     this._setTags(env, 'tags', tags)
+    this._setValue(env, 'tagsHeaderMaxLength', DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH)
+    this._setBoolean(env, 'telemetry.enabled', coalesce(
+      DD_TRACE_TELEMETRY_ENABLED, // for backward compatibility
+      DD_INSTRUMENTATION_TELEMETRY_ENABLED, // to comply with instrumentation telemetry specs
+      !this._isInServerlessEnvironment()
+    ))
+    this._setBoolean(env, 'telemetry.debug', DD_TELEMETRY_DEBUG)
+    this._setBoolean(env, 'telemetry.dependencyCollection', DD_TELEMETRY_DEPENDENCY_COLLECTION_ENABLED)
+    this._setValue(env, 'telemetry.heartbeatInterval', maybeInt(Math.floor(DD_TELEMETRY_HEARTBEAT_INTERVAL * 1000)))
+    this._setBoolean(env, 'telemetry.logCollection', coalesce(DD_TELEMETRY_LOG_COLLECTION_ENABLED, DD_IAST_ENABLED))
+    this._setBoolean(env, 'telemetry.metrics', DD_TELEMETRY_METRICS_ENABLED)
+    this._setBoolean(env, 'traceId128BitGenerationEnabled', DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED)
+    this._setBoolean(env, 'traceId128BitLoggingEnabled', DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED)
     this._setBoolean(env, 'tracing', DD_TRACING_ENABLED)
+    this._setString(env, 'version', DD_VERSION || tags.version)
   }
 
   _applyOptions (options) {
     const opts = this._options = this._options || {}
     const tags = {}
 
-    options = Object.assign({ ingestion: {} }, options, opts)
+    options = this.options = Object.assign({ ingestion: {} }, options, opts)
 
     tagger.add(tags, options.tags)
 
-    this._setString(opts, 'service', options.service || tags.service)
+    this._setValue(opts, 'appsec.blockedTemplateHtml', maybeFile(options.appsec.blockedTemplateHtml))
+    this._setValue(opts, 'appsec.blockedTemplateJson', maybeFile(options.appsec.blockedTemplateJson))
+    if (options.appsec.rules) this._setBoolean(opts, 'appsec.customRulesProvided', true)
+    this._setBoolean(opts, 'appsec.enabled', options.appsec.enabled)
+    this._setString(opts, 'appsec.obfuscatorKeyRegex', options.appsec.obfuscatorKeyRegex)
+    this._setString(opts, 'appsec.obfuscatorValueRegex', options.appsec.obfuscatorValueRegex)
+    this._setValue(opts, 'appsec.rateLimit', maybeInt(options.appsec.rateLimit))
+    if (options.appsec.rules) this._setString(opts, 'appsec.rules', options.appsec.rules)
```

Comment:
why do you need a `if (option.apsec.rules)` here ? 
```suggestion
    this._setString(opts, 'appsec.rules', options.appsec.rules)
```